mysql/datadog_checks/mysql/statements.py

```diff
@@ -184,18 +186,41 @@ def _query_summary_per_statement(self):
 
     def _normalize_queries(self, rows):
         normalized_rows = []
+        event_statements_summary_full = False
+
         for row in rows:
             normalized_row = dict(copy.copy(row))
-            try:
-                obfuscated_statement = datadog_agent.obfuscate_sql(row['digest_text'], self._obfuscate_options)
-            except Exception as e:
-                self.log.warning("Failed to obfuscate query '%s': %s", row['digest_text'], e)
-                continue
 
-            normalized_row['digest_text'] = obfuscated_statement
-            normalized_row['query_signature'] = compute_sql_signature(obfuscated_statement)
+            # A NULL digest indicates that some queries are not captured in metrics because
+            # the performance_schema query limit was hit. It is important to treat this row
+            # as a real query so that % time metrics calculations are still valid for all
+            # queries, not just tracked queries.
+            # See: https://dev.mysql.com/doc/refman/5.7/en/statement-summary-tables.html
+            if normalized_row['digest'] is None:
+                normalized_row['schema'] = VALUE_UNAVAILABLE
+                normalized_row['digest'] = VALUE_UNAVAILABLE
+                normalized_row['query'] = VALUE_UNAVAILABLE
+                normalized_row['query_signature'] = VALUE_UNAVAILABLE
+
+                event_statements_summary_full = True
+            else:
+                try:
+                    obfuscated_statement = datadog_agent.obfuscate_sql(row['digest_text'], self._obfuscate_options)
+                except Exception as e:
+                    self.log.warning("Failed to obfuscate query '%s': %s", row['digest_text'], e)
+                    continue
+
+                normalized_row['digest_text'] = obfuscated_statement
+                normalized_row['query_signature'] = compute_sql_signature(obfuscated_statement)
+
             normalized_rows.append(normalized_row)
 
+        self._check.gauge(
+            "dd.mysql.query_metrics.events_statements_summary_by_digest_full",
```

Comment:
I think we should make this a `mysql.` prefix metric and expose it to customers for them to use and monitor. 

Secondly I think we should make the metric more useful and tell customers exactly the number of lost queries which we can do by ingesting a status var; see here: https://dev.mysql.com/doc/refman/5.7/en/performance-schema-system-variables.html#sysvar_performance_schema_digests_size

Reply:
Ah, that's a great idea.